File: QID-948-MVApcabankr/MVApcabankr.r

# clear all variables
rm(list = ls(all = TRUE))
graphics.off()

# load data
x      = read.table("bank2.dat")
n      = nrow(x)
x[, 1] = x[, 1]/10
x[, 2] = x[, 2]/10
x[, 3] = x[, 3]/10
x[, 6] = x[, 6]/10

e  = eigen((n - 1) * cov(x)/n)   # calculates eigenvalues and eigenvectors and sorts them by size
e1 = e$values
x  = as.matrix(x) %*% e$vectors  # data multiplied by eigenvectors

# plot of the first vs. second PC
par(mfrow = c(2, 2))
plot(x[, 1], x[, 2], pch = c(rep(1, 100), rep(3, 100)), col = c(rep("blue", 100), 
    rep("red", 100)), xlab = "PC1", ylab = "PC2", main = "First vs. Second PC", cex.lab = 1.2, 
    cex.axis = 1.2, cex.main = 1.8)

# plot of the second vs. third PC
plot(x[, 2], x[, 3], pch = c(rep(1, 100), rep(3, 100)), col = c(rep("blue", 100), 
    rep("red", 100)), xlab = "PC2", ylab = "PC3", main = "Second vs. Third PC", cex.lab = 1.2, 
    cex.axis = 1.2, cex.main = 1.8)

# plot of the first vs. third PC
plot(x[, 1], x[, 3], pch = c(rep(1, 100), rep(3, 100)), col = c(rep("blue", 100), 
    rep("red", 100)), xlab = "PC1", ylab = "PC2", main = "First vs. Third PC", cex.lab = 1.2, 
    cex.axis = 1.2, cex.main = 1.8)

# plot of the eigenvalues
plot(e1, ylim = c(0, 2.5), xlab = "Index", ylab = "Lambda", main = "Eigenvalues of S", 
    cex.lab = 1.2, cex.axis = 1.2, cex.main = 1.8) 

File: QID-948-MVApcabankr/MVApcabankr.m
%% clear all variables and console and close windows
clear
clc
close all

%% load data  
x              = load('bank2.dat');
[n,p]          = size(x);
x(:,[1,2,3,6]) = x(:,[1,2,3,6])/10;

groups = vertcat(ones(100,1),zeros(100,1)); % Creates a vector with ones in the first 100 entries and zeros in the rest
                                            % This vector enables us to use the 'gscatter'command to plot the data in groups.

adjust = (n-1)*cov(x)/n;
[v,e]  = eigs(adjust,p,'la'); % Calculates eigenvalues and eigenvectors and sorts them by size
e1     = diag(e);             % Creates column vector of eigenvalues

% change the signs of some eigenvector. This is done only to make easier
% the comparison with R results.
v(:,[1,3,5,6]) = -v(:,[1,3,5,6]);

x = x*v;

%% plots
nr = 1:6;

%% plot of Eigenvalues
subplot(2,2,4,'FontSize',20)
scatter(nr,e1,25,'k')
xlabel('Index')
ylabel('Lambda')
title('Eigenvalues of S')
xlim([0.5 6.5])
ylim([-0.2 2.5])

%% plot of the first vs. second PC
subplot(2,2,1,'FontSize',20)
gscatter(x(:,1),x(:,2),groups,'br','+o',5,'off')
xlabel('PC1 ')
ylabel('PC2 ')
title('First vs. Second PC')

%% plot of the second vs. third PC
subplot(2,2,2,'FontSize',20)
gscatter(x(:,2),x(:,3),groups,'br','+o',5,'off')
xlabel('PC2 ')
ylabel('PC3 ')
title('Second vs. Third PC')

%% plot of the first vs. third PC
subplot(2,2,3,'FontSize',20)
gscatter(x(:,1),x(:,3),groups,'br','+o',5,'off')
xlabel('PC1 ')
ylabel('PC3 ')
title('First vs. Third PC')
